api-types/package.json

```diff
@@ -18,7 +18,7 @@
     "directory": "api-types"
   },
   "peerDependencies": {
-    "typescript": "^4.3"
+    "typescript": "5.3.3"
```

Comment:
Can this be made less restrictive so that this version doesn't need to be updated in lock step with the web app? I went with `^4.3.0 || ^5.0.0` for the client and types packages recently since I don't know of anything specific to 5.0 that is even needed for them, but we could potentially even go with `>=4.3.0`